PESDL-942 concurrent pods restart

rpdevmp opened the pull request:
PESDL-942 Concurrent pods restart (hard restart)


## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v23.3.x
- [ ] v23.2.x

## Release Notes
* none

### Improvements

* Added ability to restart all of the pods at the same time, without waiting for each pod to come up (hard restart)
* Moved some of the code into separate functions, so they could be reused and avoid code duplication
* Updated rolling restart function to also avoid code duplication
* Removed old function of nodes restart as we needed to move to pods, and created a separate function to verify basic producer_consumer that could be reused, moved it from HTT to redpanda service

NOTE: Need to perform more tests before merging

vbotbuildovich commented:
<!--RETRIED FAILURES-->
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/46546#018e5ffd-3efc-41fe-97f9-924fc0e22b19
<!--b0ee37f3-89ec-4be7-9ebd-5f8c90f17f91-->

rpdevmp commented on tests/rptest/redpanda_cloud_tests/high_throughput_test.py:
```diff
@@ -795,6 +795,7 @@ def stage_stop_wait_start(self, forced_stop: bool, downtime: int):
                    timeout_sec=restart_timeout,
                    backoff_sec=1)
 
+    @skip_if_cloud_type_is('CLOUD_TYPE_FMC')
```
Ignore this one, I will remove it

rpdevmp commented on tests/rptest/redpanda_cloud_tests/high_throughput_test.py:
```diff
@@ -808,11 +809,6 @@ def test_disrupt_cloud_storage(self):
         cloudv2 API to create/update buckets and its
         properties/policies
         """
-        if self.redpanda.cloud_type == CLOUD_TYPE_FMC:
```
Ignore this one

rpdevmp commented:
I need to update this commit/PR and remove a couple of things that got added by accident. Those were already merged.

vbotbuildovich commented:
<!--NEW FAILURES-->
new failures in https://buildkite.com/redpanda/redpanda/builds/46559#018e6188-7861-47bd-934a-666ccb3f30b6:
```
"rptest.tests.read_replica_e2e_test.TestReadReplicaService.test_identical_lwms_after_delete_records.partition_count=5.cloud_storage_type=CloudStorageType.S3"
```
<!--39fc0aa9-b26b-4bc7-99bc-d013c8c5f3b4-->

savex commented on tests/rptest/redpanda_cloud_tests/high_throughput_test.py:
```diff
@@ -1345,7 +1345,8 @@ def test_consume(self):
             self.stage_lots_of_failed_consumers()
-            self.stage_hard_restart(producer)
+            RedpandaServiceCloud.concurrent_restart_pods(producer)
+            RedpandaServiceCloud.verify_basic_produce_consume(producer)
```
My guess is this might work if there is a static id present in `globals.json`.
I suggest we update it:
```
self.redpanda.verify_basic_produce_consume(producer)
```

rpdevmp replied:
Yes, that was supposed to be self.repdanda
Updated and tested

savex reviewed (changes requested):
2 comments on business logic and 1 request for fix

rpdevmp commented:
1. To avoid code duplication, moved some of the functions from HTT to RedpandaServiceCloud so they could be reused
2. Added a concurrent restarts using threading. 1 Get a list of pods, use kubectl delete pod concurrently
3. Verify replicas, verify basic produce and consume functions
4. Tested multiple times: ducktape ... tests/rptest/redpanda_cloud_tests/high_throughput_test.py::HighThroughputTest.test_consume

rpdevmp commented on tests/rptest/services/redpanda.py:
```diff
@@ -1952,23 +1980,73 @@ def cluster_ready_replicas(cluster_name: str):
+        self.check_cluster_readiness(cluster_name, expected_replicas,
+                                     pod_timeout)
 
+    def check_cluster_readiness(self, cluster_name: str,
```
The idea to move this from test module to redpanda service, so that it could be reused by other tests as well if needed. Reusability. I updated commit and tested. We can call this inside a test.

savex reviewed (approved):
lgtm according to discussion on the sync meeting

savex reviewed (approved):
lgtm
